### PR TITLE
Fix ERC1155 token list support

### PR DESCRIPTION
**Pull request by Pantani**
# Headline
Fix ERC1155 token list support

## Problem
The filter for tokens ERC1155 are wrong for `collections/:address/collection/:token`, because Open Sea API is changed

## Solution
Change for the response parse e use slug to filter ERC1155 tokens

### CODE REVIEW

File: platform/ethereum/api.go [hunk 1]
```diff
@@ -223,17 +224,18 @@ func NormalizeCollectionPage(collections []Collection, coinIndex uint, owner str
...
-	cType := "ERC1155"
+	var symbol, version, cType, categoryAddress = "", "", "", ""
 	description := c.Description
-	categoryAddress := c.Slug
 	if len(c.Contracts) > 0 {
 		description = getValidParameter(c.Contracts[0].Description, description)
 		symbol = getValidParameter(c.Contracts[0].Symbol, symbol)
 		categoryAddress = getValidParameter(c.Contracts[0].Address, categoryAddress)
 		version = getValidParameter(c.Contracts[0].NftVersion, version)
 		cType = getValidParameter(c.Contracts[0].Type, cType)
```
Comment (madcake) [inline platform/ethereum/api.go]: More readable names.

File: platform/ethereum/api.go [hunk 2]
```diff
@@ -262,14 +264,15 @@ func NormalizeCollectiblePage(c *Collection, srcPage []Collectible, coinIndex ui
...
-	var address, externalLink = "", ""
-	cType := "ERC1155"
-	collectionID := c.Slug
+	var address, externalLink, collectionID, cType = "", "", "", ""
 	if len(c.Contracts) > 0 {
 		address = getValidParameter(c.Contracts[0].Address, address)
 		cType = getValidParameter(c.Contracts[0].Type, cType)
 		collectionID = address
 	}
+	if _, ok := slugTokens[cType]; ok {
```
Comment (madcake) [inline platform/ethereum/api.go]: What if not slug?

File: platform/ethereum/collections_client.go [hunk 3]
```diff
@@ -51,12 +51,15 @@ func (c CollectionsClient) GetCollectibles(owner string, collectibleID string) (
 		"owner": {owner},
 		"limit": {strconv.Itoa(300)},
 	}
+
 	for _, i := range collection.Contracts {
+		if _, ok := slugTokens[i.Type]; ok {
+			uriValues.Set("collection", collection.Slug)
```
Comment (madcake) [inline platform/ethereum/collections_client.go]: What if hasn't slugs?

Comment (Pantani) [reply platform/ethereum/collections_client.go]: the Blockatlas going to search by contract address

**Pull request merged**